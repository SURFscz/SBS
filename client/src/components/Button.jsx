import React from "react";
import {stopEvent} from "../utils/Utils";
import "./Button.scss";

export default function Button({onClick, txt, disabled = false, cancelButton = false,
                                   warningButton = false, className = "", icon = null}) {
    const disable = disabled ? "disabled" : "";
    const cancel = cancelButton ? "cancel" : warningButton ? "orange" : "blue";
    const cn = `button ${disable} ${cancel} ${className}`;
    return (
        <a className={cn} href={`/${encodeURIComponent(txt)}`} onClick={e => {
            stopEvent(e);
            if (!disabled) {
                onClick();
            }
        }}>{txt}{icon}</a>
    );
}