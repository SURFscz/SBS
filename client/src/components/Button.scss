@import "../stylesheets/vars.scss";

a.button {
    display: inline-block;
    padding: 10px 15px;
    background-color: $blue;
    color: white;
    border-radius: $br;
    text-transform: uppercase;
    min-width: 140px;
    text-align: center;
    cursor: pointer;
    text-decoration: none;

    svg {
        margin-left: 25px;
    }

    &:hover {
        background-color: darken($blue, 5%);
        color: darken(white, 10%);
    }

    &:hover, &:visited {
        text-decoration: none;
    }

    &.cancel, &.white {
        background-color: white;
        color: $blue;
        border-radius: $br;
        border: 1px solid $lighter-grey;
        min-width: 120px;

        &:hover {
            background-color: darken(white, 3%);
            color: darken($blue, 10%);
        }

        i {
            color: $blue;
            margin: 0;
            font-size: larger;
        }
    }

    &.delete {
        background-color: $red;
        color: white;
        border: none;

        &:hover {
            background-color: darken($red, 5%);
            color: darken(white, 10%);
        }
    }

    &.orange {
        background-color: $orange;
        color: white;
        border: none;

        &:hover {
            background-color: darken($orange, 5%);
            color: darken(white, 10%);
        }
    }

    &.green {
        background-color: $primary-green;
        color: white;
        border-radius: $br;

        &:hover {
            background-color: darken($primary-green, 5%);
            color: darken($primary-green, 10%);
        }

        i {
            color: white;
            margin: 0;
            font-size: larger;
        }
    }


    &.disabled {
        cursor: no-drop;
        background-color: $lighter-grey;
        color: white;

        &:hover {
            background-color: $lighter-grey;
            color: white;
        }
    }

    i {
        margin-left: 10px;
        color: white;
    }

}

