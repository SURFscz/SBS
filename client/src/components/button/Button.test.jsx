import React from "react";
import { render } from "@testing-library/react";
import Button from "./Button";

jest.mock("@surfnet/sds", () => ({
  Button: props => <button {...props}>{props.txt}</button>,
  ButtonSize: { Small: "small", Full: "full", Default: "default" },
  ButtonType: {
    Secondary: "secondary",
    DestructiveSecondary: "destructiveSecondary",
    Delete: "delete",
    GhostDark: "ghostDark",
    Tertiary: "tertiary",
    Primary: "primary"
  }
}));

jest.mock("../../utils/Utils", () => ({
  stopEvent: jest.fn()
}));

describe("Button", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Button onClick={jest.fn()} txt="Click me" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
