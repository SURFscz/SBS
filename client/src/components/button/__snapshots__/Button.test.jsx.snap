// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    centralize="true"
    txt="Click me"
    type="primary"
  >
    Click me
  </button>
</DocumentFragment>
`;
